lib/puppeteer/browser_runner.rb

```diff
@@ -161,14 +161,17 @@ def setup_connection(use_pipe:, timeout:, slow_mo:, preferred_revision:)
   end
 
   private def wait_for_ws_endpoint(browser_process, timeout, preferred_revision)
+    msg = []
     Timeout.timeout(timeout / 1000.0) do
-      loop do
-        line = browser_process.stderr.readline
-        /^DevTools listening on (ws:\/\/.*)$/.match(line) do |m|
-          return m[1]
+      while line = browser_process.stderr.gets
+        if /^DevTools listening on (ws:\/\/.*)$/ =~ line
+          return $1
         end
+        msg << line
       end
     end
+    raise "WS endpoint not found - process may have exited unexpectedly" \
+      " (stderr: `#{msg.map(&:strip).join " "}`)"
```

Comment:
Original puppeteer (js) uses the error message below.

```
Failed to launch the browser process! \n
(error ? ' ' + error.message : '') \n
stderr
```
https://github.com/puppeteer/puppeteer/blob/main/src/node/BrowserRunner.ts#L228

It would be better to use it also in puppeteer-ruby, unless there is special reason,

Reply:
Ah yes, definitely makes sense to mirror the js version's message. Are you okay with it being just a RuntimeError and not a documentable class?

Reply:
Yes, the error is not recoverable without rebuilding the environment. I think RuntimeError or its subclass is good.